examples/chart/teleport-kube-agent/templates/_config.tpl

```diff
@@ -33,12 +35,13 @@ kubernetes_service:
   enabled: false
   {{- end }}
 
+{{- if and (or (.Values.apps) (.Values.appResources)) (not (contains "app" (.Values.roles | toString)))}}
+  {{- fail "app role should be enabled if one of 'apps' or 'appResources' is set, see README" }}
+{{- end }}
+
 app_service:
-  {{- if contains "app" (.Values.roles | toString) }}
+  {{- if and (contains "app" (.Values.roles | toString)) (or (.Values.apps) (.Values.appResources) ($appDiscoveryEnabled)) }}
```

Comment:
```suggestion
  {{- if contains "app" (.Values.roles | toString)  }}
    {{- if not (or (.Values.apps) (.Values.appResources) ($appDiscoveryEnabled)) }}
    {{- fail "at least one of 'apps' and 'appResources' is required in chart values when app role is enabled, see README" }}
  {{- end }}
```

We should fail fast instead of hiding the error